src/partialBindTo.test.ts

```diff
@@ -0,0 +1,71 @@
+import { partialBindTo } from './partialBindTo';
```

Comment:
I want to make sure that we cover a wide range of tests, specifically I'm curious to see that everything works (both runtime and typing) with functions which take variadic parameters themselves (e.g. `(...strings: Array<string>) => strings.join(",")`) and functions that take optional arguments (e.g. `(a: string, b: isSomething = false) => isSomething ? a : "test"`)